test/tools/unified-spec-runner/match.ts

```diff
@@ -184,6 +184,18 @@ export function resultCheck(
     } else if (Long.isLong(expected) && typeof actual === 'number') {
       // Long requires special equality check
       expect(expected.equals(actual)).to.be.true;
+    } else if (Number.isNaN(actual) && Number.isNaN(expected)) {
+      // in JS, NaN isn't equal to NaN but we want to not fail if we have two NaN
+      return;
+    } else if (
+      typeof expected === 'number' &&
+      typeof actual === 'number' &&
+      expected === 0 &&
+      actual === 0
+    ) {
+      // case to handle +0 and -0
+      expect(Object.is(expected, actual)).to.be.true;
+      return;
```

Comment:
Thanks for the fix! No need to return here, just want to keep it consistent with the other blocks

Reply:
I removed the return from both of the new blocks.